### PR TITLE
Modify to support negative numbers

### PR DESCRIPTION
**Pull request by zk1991-github**
Negative numbers are supported for reactive session aging. Negative numbers indicate permanent validity

### CODE REVIEW

File: spring-session-data-redis/src/main/java/org/springframework/session/data/redis/ReactiveRedisSessionRepository.java [hunk 1]
```diff
@@ -36,7 +36,7 @@
  * A {@link ReactiveSessionRepository} that is implemented using Spring Data's
  * {@link ReactiveRedisOperations}.
  *
- * @author Vedran Pavic
+ * @author Kai Zhao
```
Comment (eleftherias) [inline spring-session-data-redis/src/main/java/org/springframework/session/data/redis/ReactiveRedisSessionRepository.java:39]: Please do not remove existing authors and instead add you name in addition.
```suggestion
 * @author Vedran Pavic
 * @author Kai Zhao
```

File: spring-session-data-redis/src/main/java/org/springframework/session/data/redis/ReactiveRedisSessionRepository.java [hunk 2]
```diff
@@ -274,8 +274,13 @@ private boolean hasChangedSessionId() {
 			String sessionKey = getSessionKey(getId());
 			Mono<Boolean> update = ReactiveRedisSessionRepository.this.sessionRedisOperations.opsForHash()
 					.putAll(sessionKey, new HashMap<>(this.delta));
-			Mono<Boolean> setTtl = ReactiveRedisSessionRepository.this.sessionRedisOperations.expire(sessionKey,
-					getMaxInactiveInterval());
+			Mono<Boolean> setTtl;
+			if (getMaxInactiveInterval().getSeconds() >= 0) {
+				setTtl = org.springframework.session.data.redis.ReactiveRedisSessionRepository.this.sessionRedisOperations.expire(sessionKey,
```
Comment (eleftherias) [inline spring-session-data-redis/src/main/java/org/springframework/session/data/redis/ReactiveRedisSessionRepository.java:279]: There is no need to add the full path for `ReactiveRedisSessionRepository`.

**Review by eleftherias — CHANGES_REQUESTED** [review]
Thanks for the PR @zk1991-github!
I have left a couple of comments inline.
In addition, please:
- Update the license header to 2021
- Add tests
- Format code using `./gradlew format`

Once you have made those changes, please squash all commits into one.

File: spring-session-data-redis/src/main/java/org/springframework/session/data/redis/ReactiveRedisSessionRepository.java (hunk 1 above)
Comment (zk1991-github) [reply spring-session-data-redis/src/main/java/org/springframework/session/data/redis/ReactiveRedisSessionRepository.java:39]: Sorry, I submit my contribution for the first time. I will modify it according to the suggestions. Thank you

File: spring-session-data-redis/src/main/java/org/springframework/session/data/redis/ReactiveRedisSessionRepository.java (hunk 2 above)
Comment (zk1991-github) [reply spring-session-data-redis/src/main/java/org/springframework/session/data/redis/ReactiveRedisSessionRepository.java]: I have completed the modification as required, please review

**Comment by zk1991-github** [comment]
> 感谢公关@zk1991-github！
> 我已经在内联留下了一些评论。
> 此外，请：
> 
> * 将许可证标题更新为 2021
> * 添加测试
> * 使用格式化代码 `./gradlew format`
> 
> 完成这些更改后，请将所有提交合并为一个。

**Comment by zk1991-github** [comment]
Sorry, I closed PR accidentally. Please review it again

**Comment by zk1991-github** [comment]
There are some problems that have not been completely modified. I will close this PR and resubmit a new PR. thank you

**Pull request closed without merging**